ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/services/RefragmentationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.entities.LdesFragment;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.tree.member.entities.Member;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.tree.member.repository.MemberRepository;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+import org.springframework.stereotype.Service;
+
+import java.util.stream.Stream;
+
+@Service
+public class RefragmentationServiceImpl implements RefragmentationService {
+	private final MemberRepository memberRepository;
+	private final ObservationRegistry observationRegistry;
+
+	public RefragmentationServiceImpl(MemberRepository memberRepository, ObservationRegistry observationRegistry) {
+		this.memberRepository = memberRepository;
+		this.observationRegistry = observationRegistry;
+	}
+
+	// TODO refragmentation of old members might take a while. During this period
+	// new arrived members will not be fragmented.
```

Comment:
What happens in the current setup if a new member is ingested while we are still processing this?

Reply:
That new member won't be fragmented in this view. Since the Refragmentation happens before the ViewCollection of the fragmentation process is updated.
That's actually what the remark is about: There's a time window in which you are busy refragmenting old members, but in which you won't process nor queue newly ingested members.

When we revise the fragmentation process, we should take this into account, but in my opinion it would be too much of a change to also include this in the PR. Moreover, since the discussing on horizontal scaling are still ongoing.